### ISSUE TITLE
incorrect resolution of the webcam stream

### ISSUE THREAD

**Issue by lavrentijav** [ultralytics/ultralytics#21545]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

I noticed an error where when using a webcam stream of any resolution. it always transforms into something else.
```
1/1: 1... Success  (inf frames of shape 640x480 at 30.00 FPS)

0: 384x480 (no detections), 394.0ms
0: 384x480 (no detections), 33.8ms
```
the current webcam resolution is 1920 x 1080

The 640x480 resolution is used even when using the dynamic onnx model or enabling the rect parameter.

### Environment

```
Ultralytics 8.3.170  Python-3.12.6 torch-2.7.1+cu118 CUDA:0 (NVIDIA GeForce GTX 1080 with Max-Q Design, 8192MiB)
Setup complete  (12 CPUs, 15.9 GB RAM, 451.1/476.3 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.12.6
Install             pip
Path                C:\Users\Public\Projects\CS_AIM.zip\.venv\Lib\site-packages\ultralytics
RAM                 15.92 GB
Disk                451.1/476.3 GB
CPU                 Intel Core(TM) i7-8750H 2.20GHz
CPU count           12
GPU                 NVIDIA GeForce GTX 1080 with Max-Q Design, 8192MiB
GPU count           1
CUDA                11.8

numpy                2.1.2>=1.23.0
matplotlib           3.10.3>=3.3.0
opencv-python        4.12.0.88>=4.6.0
pillow               11.0.0>=7.1.2
pyyaml               6.0.2>=5.3.1
requests             2.32.4>=2.23.0
scipy                1.16.1>=1.4.1
torch                2.7.1+cu118>=1.8.0
torch                2.7.1+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision          0.22.1+cu118>=0.9.0
tqdm                 4.67.1>=4.64.0
psutil               7.0.0
py-cpuinfo           9.0.0
pandas               2.3.1>=1.1.4
ultralytics-thop     2.0.14>=2.0.0

```

### Minimal Reproducible Example

```
from ultralytics import YOLO
model = YOLO("yolo12n.pt")
source = 1
results = model(source=source, stream=True, rect=True)

for result in results:
    print(result.orig_img.shape)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lavrentijav, thank you for reporting this issue and for providing detailed environment information and a reproducible example! 🚀 This is an automated response to help you get the fastest support. An Ultralytics engineer will also review your report and assist you soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for answers to common questions and configuration details. Since this appears to be a 🐛 Bug Report, thank you for already including a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)—this will help us debug the resolution issue effectively.

If you haven't already, please ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

YOLO can be run in various verified environments (with dependencies such as [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For real-time support or to connect with the Ultralytics community, join us on [Discord](https://discord.com/invite/ultralytics) 🎧, [Discourse](https://community.ultralytics.com/), or [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the badge above is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

Thank you for your contribution and patience while we look into this! 🛠️

**Comment by Y-T-G** [comment]
It shows the input resolution after being resized for inference which is not the same as stream resolution

**Comment by lavrentijav** [comment]
Then how do I output in the required resolution. How do I process the stream in its original resolution?

**Comment by Y-T-G** [comment]
You typically don't use original resolution for inference, otherwise the inference would be a lot slower. The output is still in original resolution.

The pretrained models use `imgsz=640`, so you will have to stick to that for best accuracy. You can increase the `imgsz`, but that will probably degrade accuracy since the model wasn't trained at that `imgsz`.

https://docs.ultralytics.com/modes/predict/#inference-arguments
